controllers/awsmachine_controller.go

```diff
@@ -510,17 +577,7 @@ func (r *AWSMachineReconciler) deleteEncryptedBootstrapDataSecret(machineScope *
 	return nil
 }
 
-func (r *AWSMachineReconciler) getOrCreate(scope *scope.MachineScope, ec2svc services.EC2MachineInterface, secretSvc services.SecretsManagerInterface) (*infrav1.Instance, error) {
-	instance, err := r.findInstance(scope, ec2svc)
```

Comment:
Might be I'm missing something, but I don't see strong reason to move findInstance in reconcileNormal vs implementing here the conditions and the extra patching logic, but I leave the final call to @detiber on this

Reply:
I'm not overly opinionated on where we call `findInstance` as long as we are properly guarding against eventual consistency bugs, such as the potential one I described in https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1740#discussion_r440444519